### PR TITLE
feat(sampiler): initial version of sample transpiler

### PR DESCRIPTION
**Pull request by rix0rrr**
This is the initial version of a tool that will parse the TypeScript
samples and emit equivalent sample code in Python.

For now, its use is limited:

- only supports Python
- doesn't support properly integrating/compiling with a multi-package
  build yet.

However, it will be an initial version we can demo and iterate on.

For a slightly outdated example of generated code, see: https://gist.github.com/rix0rrr/7c4a651f8eef1e94b006059209e805dd

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: packages/jsii-sampiler/bin/jsii-snippet [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env node
+require('./jsii-snippet.js');
```
Comment (eladb) [inline packages/jsii-sampiler/bin/jsii-snippet]: Rename to jsii-samplier I guess

File: packages/jsii-sampiler/README.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
+# jsii-sampiler: a transpiler for code samples
+
+Utility to transcribe example code snippets from TypeScript to other
+jsii languages.
+
+Has knowledge about jsii language translation conventions to do the
+translations. Only supports a limited set of TypeScript language features.
+
```
Comment (eladb) [inline packages/jsii-sampiler/README.md]: Describe the usage of the tool

File: packages/jsii-sampiler/test/python/expressions.test.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  'console.log(`Hello, ${x}, it works ${y}!`);',
+  ].join('\n'), `
+  x = "world"
+  y = "well"
+  print(f"Hello, {x}, it works {y}!")
+  `);
+});
+
+test('non-null expression', () => {
+  expectPython([
```
Comment (eladb) [inline packages/jsii-sampiler/test/python/expressions.test.ts]: Maybe prepare for the next language: ```expect(input, { python: output })```

File: packages/jsii-sampiler/package.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "jsii-sampiler",
+  "version": "0.16.0",
+  "description": "Translate TypeScript code snippets to other languages",
+  "main": "lib/index.js",
+  "bin": {
```
Comment (eladb) [inline packages/jsii-sampiler/package.json]: Fix

File: packages/jsii-sampiler/test/python/expressions.test.ts (hunk 3 above)
Comment (rix0rrr) [reply packages/jsii-sampiler/test/python/expressions.test.ts]: Yep, totally planning to do that. It's the only way to cover all features across all languages.

**Pull request merged**